### ISSUE TITLE
Same setup - different training loss (outcome)

### ISSUE THREAD

**Issue by keck91** [BVLC/caffe#6588]
Dear community,

I currenty face the problem, that the performance of my training varies strongly.
Sometimes, I have a usable caffemodel after training and sometimes not.
So I started a test. I picked 100 labeled images and run 3 short trainings with it, without changing anything (same data set, same learning rate of 0.001).
Results:
First run, "Radar09" with 50,000 iterations resulted in a loss of 0.001080. After 1,000 iterations, the loss was under 1.0
Second run, "Radar10" with 2.500 iterations resulted in a loss of 8.561
Third run, "Radar11" with 2.500 iterations resulted in a loss of 0.5212
Fourth run "Radar12" with 2.500 iterations resulted in a loss of 8.909

Why is this so? 
What could be the reason? 
How could I avoid this?
For a noticeably larger dataset and more iterations (3,000 images and 100,000 iterations) this will be a problem.

![radar09_crop](https://user-images.githubusercontent.com/42660731/47701608-b7e35b80-dc5d-11e8-9578-3355daf4ff7d.png)
![radar10](https://user-images.githubusercontent.com/42660731/47701619-c03b9680-dc5d-11e8-87a4-1807e4f4a377.png)
![radar11](https://user-images.githubusercontent.com/42660731/47701624-c3368700-dc5d-11e8-8fcc-0602c9fc7d02.png)
![radar12](https://user-images.githubusercontent.com/42660731/47703353-54f4c300-dc63-11e8-9fa3-9cd461c058f7.png)

Thank you very much! :)

### System configuration

* Operating system: Ubuntu 16.04
* CUDA version (if applicable): 8.0
* CUDNN version (if applicable): 5.1
* Python version (if using pycaffe): 2.7

**Comment by csukuangfj** [comment]
One possible reason might be the seed for the random number generator.

See the [code][1] below
```proto
  // If non-negative, the seed with which the Solver will initialize the Caffe
  // random number generator -- useful for reproducible results. Otherwise,
  // (and by default) initialize using a seed derived from the system clock.
  optional int64 random_seed = 20 [default = -1];
```

[1]: https://github.com/BVLC/caffe/blob/99bd99795dcdf0b1d3086a8d67ab1782a8a08383/src/caffe/proto/caffe.proto#L210

**Comment by keck91** [comment]
Hello csukuangfj,

Thank you for your answer. I checked my src/caffe/proto/caffe.proto.
optional int64 random_seed is already set to 20

![screenshot from 2018-11-06 15-08-00](https://user-images.githubusercontent.com/42660731/48045950-c8538300-e1d5-11e8-80dd-52f928f77bd5.png)

**Comment by csukuangfj** [comment]
@keck91 
20 is an id that used by protobuf.

you should add
```proto
random_seed: 20
```
to your solver prototxt to set it to 20, otherwise, it is -1 by default.

**Comment by keck91** [comment]
Ah. I see!
Thank you.
I will continue with random_seed: 20 from now on, but unfortunately I don't know, which seed it was for the "vital" trainings not stucking at 6.9

**Comment by ABCDAS** [comment]
[libprotobuf ERROR C:\Users\guillaume\work\caffe-builder\build_v140_x64\packages\protobuf\protobuf_download-prefix\src\protobuf_download\src\google\protobuf\message_lite.cc:248] Exceeded maximum protobuf size of 2GB.
F0814 15:05:56.169399 11676 io.cpp:78] Check failed: proto.SerializeToOstream(&output)
*** Check failure stack trace: ***

do you solve this question?
thank you

**Comment by csukuangfj** [comment]
@ABCDAS 

Google is your best friend.

https://github.com/BVLC/caffe/issues/2006
